### ISSUE TITLE
Distinguish between education destinations

### ISSUE THREAD

**Issue by sebhoerl** [eqasim-org/ile-de-france#108]
Currently, we do not distinguish between education facility types for different population strata. The following would need to be done:

- When processing the BPE (`data.bpe.cleaned`), write out the `TYPEQU` holding the facility type. See https://www.insee.fr/fr/statistiques/fichier/3568638/Contenu_bpe20_ensemble_xy.pdf
- When creating education facilities (`synthesis.locations.education`), use this attribute to distinguish between `N` education facility types. These `N` need to be defined in some way. This could either be via age class or cross-over with other attributes (highest diploma / current diploma)
- These `N` classes (if more complex than age) should then be implemented in an additional stage or in the `synthesis.population.enriched` stage (or directly in the location assignment, see below)
- We need to preprocess the OD matrix for education. Currently, it only gives a probability for each destination municipality, given an origin municipality. What we need is to have `N` matrices, with zeros for destination municipalities where the selected category `n` is not available. *In total* the `N` matrices should then give the original matrix, so some decomposition needs to be applied here. This needs a new stage, potentially in `synthesis.population.spatial.primary.education_matrices`.
- Assignment of facilities happens in two steps. First, we take the OD matrix and then assign an education destination municipality to each person in `synthesis.population.spatial.primiary.candidates`. Here, we need to perform the assignment for each of the `N` categories individually (i.e. select all persons with category `n`, then select the matrix for category `n` and perform the assignment as described in the paper / in the code already)
- After, assignment of a facility from the selected municipalities happens. This does not need much modification in `synthesis.population.spatial.primary.locations`, only the assignment needs to happen per category `n`.

**Comment by Nitnelav** [comment]
is this fixed in https://github.com/eqasim-org/ile-de-france/pull/252 ?

**Comment by sebhoerl** [comment]
I would keep it open and see if @MarieMcLaurent could include the cleaning / processing of the education ministry data into the pipeline instead of providing an external JSON?

**Comment by MarieMcLaurent** [comment]
Yes, I could include the cleaning / processing of the education ministry data and update `data.external.education`. I would need to see how to adapt my code that process it to fit eqasim pipeline. Would it be necessary to get rid at the same time of possibility to provide an external JSON or it can still be include ?

**Comment by sebhoerl** [comment]
We could still keep both options

**Comment by MarieMcLaurent** [comment]
With update of BPE and in comparison with education ministry data, we found that numbers of students are more complete in BPE data. So instead of providing default weight, weights refer now to this numbers. Since we are still allowing use of external file, it may not be useful to implement processing of education ministry data (already include in BPE and not always fully available) unless we would need to have an update version other than 2023 data.
